src/dag_node.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright 2019-2020 libsbn project contributors.
+// libsbn is free software under the GPLv3; see LICENSE file for details.
+
+#include "dag_node.hpp"
+
+std::string GetNeighborString(std::vector<size_t> neighbors)
```

Comment:
This is probably not worth fiddling with now, but have you tried using the prettyprint.hpp stuff? You can take stl containers and just << them to cout. I expect that you could << them to an ostringstream too.